lib/lib-table-cluster/src/main/java/org/sagebionetworks/table/cluster/TableIndexDAOImpl.java

```diff
@@ -934,6 +938,37 @@ public void createViewSnapshotFromEntityReplication(ObjectType objectType, Long
 			outputStream.writeNext(row);
 		});
 	}
+	
+	// Translates the Column Model schema into a column metadata schema, that maps to the object/annotation replication index
+	List<ColumnMetadata> translateSchema(ObjectType objectType, List<ColumnModel> schema) {
+		return schema.stream()
+			.map((ColumnModel columnModel) -> translateColumnModel(objectType, columnModel))
+			.collect(Collectors.toList());
+	}
+	
+	ColumnMetadata translateColumnModel(ObjectType objectType, ColumnModel model) {
+		// First determine if this an object column or an annotation
+		Optional<EntityField> entityField = findObjectField(objectType, model);
+		
+		String selectColumnForId = SQLUtils.getColumnNameForId(model.getId());
+		boolean isObjectReplicationField;
+		String selectColumnName;
+		
+		if (entityField.isPresent()) {
+			isObjectReplicationField = true;
+			selectColumnName = entityField.get().getDatabaseColumnName();
+		} else {
+			isObjectReplicationField = false;
+			selectColumnName = SQLUtils.translateColumnTypeToAnnotationValueName(model.getColumnType());
+		}
+		
+		return new ColumnMetadata(model, selectColumnName, selectColumnForId, isObjectReplicationField);
+	}
+	
+	Optional<EntityField> findObjectField(ObjectType objectType, ColumnModel model) {

```

Comment:
Convert EntityField to be more generic -> ObjectField, and use it statically in many places, but not all.